physics/GFS_rrtmgp_thompsonmp_pre.F90

```diff
@@ -147,11 +147,11 @@ subroutine GFS_rrtmgp_thompsonmp_pre_run(nCol, nLev, nTracers, ncnd, doSWrad, do
     ! Cloud particle sizes and number concentrations...
     
     ! First, prepare cloud mixing-ratios and number concentrations for Calc_Re
-    rho  = p_lay(1:nCol,1:nLev)/(con_rd*t_lay(1:nCol,1:nLev))    
-    orho = 1./rho    
     do iLay = 1, nLev
        do iCol = 1, nCol    
           qv_mp(iCol,iLay) = q_lay(iCol,iLay)/(1.-q_lay(iCol,iLay))
+          rho(iCol,iLay)  = 0.622*p_lay(1:nCol,1:nLev)/(con_rd*t_lay(1:nCol,1:nLev)*(qv_mp(iCol,iLay)+0.622))    
```

Comment:
To save some memory, shouldn't we be using scalar (singular) values inside the loops not arrays?  Unless we are saving them for later usage, in which case I guess it's necessary.

Reply:
I think the loops in lines 149-166 and in lines 169-178 can be combined, which would allow us to make rho and orho scalars.

Reply:
I added here just one line at request from Dustin Swales.

Reply:
Note that this line is also incorrect, because it multiplies arrays `p_lay(1:nCol,1:nLev)` and `t_lay(1:nCol,1:nLev)`, i.e. entire 2-dim arrays, with a scalar `qv_mp(iCol,iLay)` and assigns it to a scalar `rho(iCol,iLay)`. This leads to compile errors in debug mode. I am going to fix this in the PR I am preparing for @tanyasmirnova.

Reply:
Of course! A copy/paste problem Thank you for fixing this.